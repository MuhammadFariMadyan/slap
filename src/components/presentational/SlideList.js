import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import "typeface-roboto";

import Slide from "./Slide";

const Container = styled.div`
  width: 220px;
  border-right: 1px solid #ccc;
  background: #eee;
  overflow-y: auto;
`;

const SlideList = ({ className, items, active, onClick, onClickDelete }) => (
  <Container className={className}>
    {items.map((item, index) => (
      <Slide
        key={index}
        index={index + 1}
        {...item}
        onClick={() => onClick(item)}
        onClickDelete={() => onClickDelete(item)}
        isActive={active === item.id}
      />
    ))}
  </Container>
);

SlideList.propTypes = {
  className: PropTypes.string,
  items: PropTypes.array,
  active: PropTypes.string,
  onClick: PropTypes.func,
  onClickDelete: PropTypes.func
};

SlideList.defaultProps = {
  items: []
};

export default SlideList;
