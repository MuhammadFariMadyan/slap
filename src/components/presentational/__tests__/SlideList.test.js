import React from "react";
import { shallow } from "enzyme";

import SlideList from "../SlideList";

jest.mock("../Slide", () => "Slide");

describe("SlideList", () => {
  const setup = propOverrides => {
    const props = {
      items: [
        {
          id: "slide-1",
          projectID: "project-1",
          contents: [
            {
              id: "content-1",
              slideID: "slide-1",
              projectID: "project-1",
              type: "text",
              content: "Stay hungry",
              position: {
                x: 128,
                y: 96
              }
            }
          ]
        },
        {
          id: "slide-2",
          projectID: "project-1",
          contents: [
            {
              id: "content-2",
              slideID: "slide-2",
              projectID: "project-1",
              type: "text",
              content: "Stay foolish",
              position: {
                x: 128,
                y: 128
              }
            }
          ]
        }
      ],
      onClick: jest.fn(),
      onClickDelete: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<SlideList {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders with active item", () => {
    const { wrapper } = setup({ active: "slide-2" });
    const slide = wrapper.find("Slide");

    expect(slide.at(0).props().isActive).toEqual(false);
    expect(slide.at(1).props().isActive).toEqual(true);
  });

  it("should calls 'onClick' callback while item clicked", () => {
    const { wrapper, props } = setup();

    props.items.forEach((item, index) => {
      wrapper
        .find("Slide")
        .at(index)
        .props()
        .onClick(item);

      expect(props.onClick).toBeCalledWith(item);
    });
  });

  it("should calls 'onClickDelete' callback while item delete clicked", () => {
    const { wrapper, props } = setup();

    props.items.forEach((item, index) => {
      wrapper
        .find("Slide")
        .at(index)
        .props()
        .onClickDelete(item);

      expect(props.onClickDelete).toBeCalledWith(item);
    });
  });
});
