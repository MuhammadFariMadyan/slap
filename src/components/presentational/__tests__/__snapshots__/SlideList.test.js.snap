// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideList should renders default 1`] = `
<styled.div>
  <Slide
    contents={
      Array [
        Object {
          "content": "Stay hungry",
          "id": "content-1",
          "position": Object {
            "x": 128,
            "y": 96,
          },
          "projectID": "project-1",
          "slideID": "slide-1",
          "type": "text",
        },
      ]
    }
    id="slide-1"
    index={1}
    isActive={false}
    key="0"
    onClick={[Function]}
    onClickDelete={[Function]}
    projectID="project-1"
  />
  <Slide
    contents={
      Array [
        Object {
          "content": "Stay foolish",
          "id": "content-2",
          "position": Object {
            "x": 128,
            "y": 128,
          },
          "projectID": "project-1",
          "slideID": "slide-2",
          "type": "text",
        },
      ]
    }
    id="slide-2"
    index={2}
    isActive={false}
    key="1"
    onClick={[Function]}
    onClickDelete={[Function]}
    projectID="project-1"
  />
</styled.div>
`;
